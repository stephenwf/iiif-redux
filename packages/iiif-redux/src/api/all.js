const getAllCollections = state => state.resources.collections;
const getAllExternalResources = state => state.resources.externalResources;
const getAllLayers = state => state.resources.layers;
const getAllServices = state => state.resources.services;
const getAllManifests = state => state.resources.manifests;
const getAllSequences = state => state.resources.sequences;
const getAllRanges = state => state.resources.ranges;
const getAllCanvases = state => state.resources.canvases;
const getAllCanvasReferences = state => state.resources.canvasReferences;
const getAllImages = state => state.resources.imageResources;
const getAllAnnotations = state => state.resources.annotations;
const getAllAnnotationLists = state => state.resources.annotationLists;
const getAllAnnotationPages = state => state.resources.annotationPages;
const getAllAnnotationCollections = state =>
  state.resources.annotationCollections;
const getAllContentResources = state => state.resources.contentResources;
const getAllResources = state => state.resources;

export {
  getAllCollections,
  getAllExternalResources,
  getAllLayers,
  getAllServices,
  getAllManifests,
  getAllSequences,
  getAllRanges,
  getAllCanvases,
  getAllCanvasReferences,
  getAllImages,
  getAllAnnotationLists,
  getAllAnnotationPages,
  getAllAnnotationCollections,
  getAllAnnotations,
  getAllContentResources,
  getAllResources,
};
