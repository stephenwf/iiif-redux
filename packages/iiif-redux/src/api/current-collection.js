import { getCurrentCollection } from './current';
import collection from './collection';

const {
  // Technical
  getId,
  getType,
  getViewingHint,
  getNavDate,
  // Descriptive
  getLabel,
  getDescription,
  getMetadata,
  getAttribution,
  getLogo,
  getLicense,
  getThumbnailId,
  getThumbnail,
  // Linking
  getSeeAlsoIds,
  getSeeAlso,
  getServiceIds,
  getService,
  getRelatedIds,
  getRelated,
  getRenderingIds,
  getRendering,
  getWithinIds,
  getWithin,
  // Paging
  getFirstId,
  getFirst,
  getLastId,
  getLast,
  getTotal,
  getNextId,
  getNext,
  getPreviousId,
  getPrevious,
  getStartIndex,
  // Structural
  getCollectionIds,
  getCollections,
  getManifestIds,
  getManifests,
  getMemberIds,
  getMembers,
  getOtherContentIds,
  getOtherContent,
} = collection(getCurrentCollection);

export {
  // Technical
  getId,
  getType,
  getViewingHint,
  getNavDate,
  // Descriptive
  getLabel,
  getDescription,
  getMetadata,
  getAttribution,
  getLogo,
  getLicense,
  getThumbnailId,
  getThumbnail,
  // Linking
  getSeeAlsoIds,
  getSeeAlso,
  getServiceIds,
  getService,
  getRelatedIds,
  getRelated,
  getRenderingIds,
  getRendering,
  getWithinIds,
  getWithin,
  // Paging
  getFirstId,
  getFirst,
  getLastId,
  getLast,
  getTotal,
  getNextId,
  getNext,
  getPreviousId,
  getPrevious,
  getStartIndex,
  // Structural
  getCollectionIds,
  getCollections,
  getManifestIds,
  getManifests,
  getMemberIds,
  getMembers,
  getOtherContentIds,
  getOtherContent,
  // Other
  getCurrentCollection,
};
