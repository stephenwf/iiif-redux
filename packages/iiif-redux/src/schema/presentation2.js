/**
 * Presentation 2 - schema definition
 * ============================================================================
 *
 * - Collection
 * - Manifest
 * - Sequence
 * - Canvas
 * - Annotation
 * - AnnotationList
 * - Range
 * - Layer
 * - Image Content (Annotation)
 * - Other Content (Annotation List)
 *
 * - External resources
 * - Services
 */

// @todo Image Service API.
import { compose } from 'redux';
import { schema, normalize } from 'normalizr';
import moveStartCanvasToSequence from '../compat/moveStartCanvasToSequence';
import preprocessLinkedEntities from '../compat/preprocessLinkedEntities';
import addMissingIds from '../compat/addMissingIds';

function createEntity(name) {
  const options = {
    idAttribute: '@id',
  };

  return new schema.Entity(name, {}, options);
}

const collection = createEntity('collections');
const manifest = createEntity('manifests');
const sequence = createEntity('sequences');
const canvas = createEntity('canvases');
const annotation = createEntity('annotations');
const annotationList = createEntity('annotationLists');
const range = createEntity('ranges');
const layer = createEntity('layers');
const imageResource = createEntity('imageResources'); // thumbnails + image service
const choice = createEntity('choices');

// Unofficial types?
const externalResource = createEntity('externalResources');
const service = createEntity('services', true);
const within = new schema.Array(
  {
    collection,
    manifest,
    externalResource,
    layer,
  },
  entity => {
    switch (entity['@type']) {
      case 'sc:Collection':
        return 'collection';
      case 'sc:Manifest':
        return 'manifest';
      case 'sc:Layer':
        return 'layer';
      default:
        return 'externalResource';
    }
  }
);

const rangeMember = new schema.Union(
  {
    canvas,
    range,
  },
  input => (input['@type'] === 'sc:Canvas' ? 'canvas' : 'range')
);

/**
 * Dereferencable resources:
 *
 * - Collection (Required)
 * - Manifest (Required)
 * - First Sequence (Optional)
 * - Nth Sequence (Required)
 * - Canvas (Optional)
 * - Annotation (Recommended)
 * - AnnotationList (Required)
 * - Range (Optional)
 * - Layer (Optional)
 * - Image Content (Required)
 * - Other Content (Required)
 */
const RESOURCE_TYPE_MAP = {
  'sc:Collection': 'collection',
  'sc:Sequence': 'sequence',
  'sc:Manifest': 'manifest',
  'sc:Canvas': 'canvas',
  'sc:AnnotationList': 'annotationList',
  'sc:Annotation': 'annotation',
  'sc:Range': 'range',
  'sc:Layer': 'layer',
  'dctypes:Image': 'imageResource',
};
const resource = new schema.Union(
  {
    collection,
    sequence,
    manifest,
    canvas,
    annotationList,
    annotation,
    range,
    layer,
    imageResource,
    service,
  },
  entity => {
    if (RESOURCE_TYPE_MAP[entity['@type']]) {
      return RESOURCE_TYPE_MAP[entity['@type']];
    }
    if (entity.profile) {
      return 'service';
    }
    throw Error('Unknown entity type');
  }
);

// ===========================================================================
// 1) Collections
// ===========================================================================
collection.define({
  collections: new schema.Array(
    {
      collection,
      manifest,
    },
    entity => (entity['@type'] === 'sc:Manifest' ? 'manifest' : 'collection')
  ),
  manifests: [manifest],
  members: new schema.Array(
    {
      collection,
      manifest,
    },
    entity => (entity['@type'] === 'sc:Manifest' ? 'manifest' : 'collection')
  ),

  // Linking
  seeAlso: [externalResource],
  service: [service],
  related: [externalResource],
  rendering: [externalResource],
  otherContent: [annotationList],
  within: within,

  // Extra
  thumbnail: imageResource,
});

// ===========================================================================
// 2) Manifests
// ===========================================================================
manifest.define({
  sequences: [sequence],
  structures: [range],

  // Linking
  seeAlso: [externalResource],
  service: [service],
  related: [externalResource],
  rendering: [externalResource],
  otherContent: [annotationList],
  within: within,

  // Extra
  thumbnail: imageResource,
});

// ===========================================================================
// 3) Sequences
// ===========================================================================
sequence.define({
  canvases: [canvas],

  // Linking
  seeAlso: [externalResource],
  service: [service],
  related: [externalResource],
  rendering: [externalResource],
  otherContent: [annotationList],
  within: within,
  startCanvas: canvas,

  // Extra
  thumbnail: imageResource,
});

// ===========================================================================
// 4) Canvases
// ===========================================================================
canvas.define({
  images: [annotation],

  // Linking
  seeAlso: [externalResource],
  service: [service],
  related: [externalResource],
  rendering: [externalResource],
  otherContent: [annotationList],
  within: within,

  // Extra
  thumbnail: imageResource,
});

// ===========================================================================
// 5) Annotation
// ===========================================================================
annotation.define({
  resource: new schema.Union(
    {
      imageResource,
      externalResource,
      choice,
    },
    input => {
      switch (input['@type']) {
        case 'dctypes:Image':
          return 'imageResource';
        case 'oa:Choice':
          return 'choice';
        default:
          return 'externalResource';
      }
    }
  ),
  on: canvas,

  // Linking
  seeAlso: [externalResource],
  service: [service],
  related: [externalResource],
  rendering: [externalResource],
  within: within,

  // Extra
  thumbnail: imageResource,
});

// ===========================================================================
// 6) Annotation List
// ===========================================================================
annotationList.define({
  resources: [annotation],

  // Linking
  seeAlso: [externalResource],
  service: [service],
  related: [externalResource],
  rendering: [externalResource],
  within: within,

  // Extra
  thumbnail: imageResource,
});

// ===========================================================================
// 7) Range
// ===========================================================================
range.define({
  resources: [annotation],
  members: [rangeMember],
  canvases: [canvas],
  ranges: [range],

  // Linking
  seeAlso: [externalResource],
  service: [service],
  related: [externalResource],
  rendering: [externalResource],
  within: within,
  startCanvas: canvas,
  contentLayer: layer,

  // Extra
  thumbnail: imageResource,
});

// ===========================================================================
// 8) Layer
// ===========================================================================
layer.define({
  // Linking
  seeAlso: [externalResource],
  service: [service],
  related: [externalResource],
  rendering: [externalResource],
  within: within,
});

// ===========================================================================
// 9) Image resource
// ===========================================================================
imageResource.define({
  service: [service],
});

// ===========================================================================
// 10) Choice
// ===========================================================================
choice.define({
  default: annotation,
  item: [annotation],
});

const preprocess = compose(
  moveStartCanvasToSequence,
  preprocessLinkedEntities,
  addMissingIds
);

const normalizeResource = (rawResource, customSchema = resource) =>
  normalize(preprocess(rawResource), customSchema);

export {
  normalizeResource as normalize,
  preprocess,
  resource,
  collection,
  manifest,
  sequence,
  canvas,
  annotation,
  annotationList,
  range,
  layer,
  imageResource,
  externalResource,
  within,
  service,
};
