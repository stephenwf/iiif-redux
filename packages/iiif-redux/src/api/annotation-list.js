import { getAllAnnotationLists } from './all';
import byIdSelectorFactory from '../utility/byIdSelectorFactory';
import resourceListSelectorFactory from '../utility/resourceListSelectorFactory';
import presentation2annotationList from './2.x/annotation-list';

const annotationList = presentation2annotationList;

export default annotationList;

export const annotationLists = resourceListSelectorFactory(
  getAllAnnotationLists,
  annotationList
);

export const annotationListByIdSelector = byIdSelectorFactory(
  annotationList,
  'annotationLists'
);
