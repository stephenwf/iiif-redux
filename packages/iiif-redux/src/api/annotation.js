import byIdSelectorFactory from '../utility/byIdSelectorFactory';
import resourceListSelectorFactory from '../utility/resourceListSelectorFactory';
import presentation2Annotation from './2.x/annotation';
import { getAllAnnotations } from './all';

const annotation = presentation2Annotation;

export default annotation;

export const annotations = resourceListSelectorFactory(
  getAllAnnotations,
  annotation
);

export const annotationByIdSelector = byIdSelectorFactory(
  annotation,
  'annotations'
);
