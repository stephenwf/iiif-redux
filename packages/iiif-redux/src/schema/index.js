import * as presentation2 from './presentation2';
import * as presentation3 from './presentation3';

export { presentation2, presentation3 };

// Default is 2 until 3 is ratified?
export default presentation2;
