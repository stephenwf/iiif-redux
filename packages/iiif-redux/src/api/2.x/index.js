import annotationList, {
  mappings as annotationListMapping,
} from './annotation-list';
import annotation, { mappings as annotationMapping } from './annotation';
import canvas, { mappings as canvasMapping } from './canvas';
import collection, { mappings as collectionMapping } from './collection';
import externalResource, {
  mappings as externalResourceMapping,
} from './external-resource';
import imageResource, {
  mappings as imageResourceMapping,
} from './image-resource';
import manifest, { mappings as manifestMapping } from './manifest';
import range, { mappings as rangeMapping } from './range';
import sequence, { mappings as sequenceMapping } from './sequence';

export const mappings = {
  annotationList: annotationListMapping,
  annotation: annotationMapping,
  canvas: canvasMapping,
  collection: collectionMapping,
  externalResource: externalResourceMapping,
  imageResource: imageResourceMapping,
  manifest: manifestMapping,
  range: rangeMapping,
  sequence: sequenceMapping,

  // Aliases.
  annotationPage: annotationListMapping,
  contentResources: imageResourceMapping, // @todo change when image/external get merged.
};

export {
  annotationCollection,
  annotationList,
  annotation,
  canvas,
  collection,
  externalResource,
  imageResource,
  manifest,
  range,
  sequence,
};
