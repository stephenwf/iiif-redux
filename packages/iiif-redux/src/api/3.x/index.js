import annotationCollection, {
  mappings as annotationCollectionMapping,
} from './annotation-collection';
import annotationPage, {
  mappings as annotationPageMapping,
} from './annotation-page';
import annotation, { mappings as annotationMapping } from './annotation';
import canvas, { mappings as canvasMapping } from './canvas';
import collection, { mappings as collectionMapping } from './collection';
import contentResource, {
  mappings as contentResourceMapping,
} from './content-resource';
import manifest, { mappings as manifestMapping } from './manifest';
import range, { mappings as rangeMapping } from './range';

export const mappings = {
  annotationCollection: annotationCollectionMapping,
  annotationPage: annotationPageMapping,
  annotation: annotationMapping,
  canvas: canvasMapping,
  collection: collectionMapping,
  contentResource: contentResourceMapping,
  manifest: manifestMapping,
  range: rangeMapping,
};

export {
  annotationCollection,
  annotationPage,
  annotation,
  canvas,
  collection,
  contentResource,
  manifest,
  range,
};
