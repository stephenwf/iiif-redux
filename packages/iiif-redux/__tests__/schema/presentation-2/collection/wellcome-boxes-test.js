import { normalize } from 'normalizr';
import { collection } from '../../../../src/schema/presentation2';
import boxes from '../../../fixtures/wellcome-boxes';
import * as currentCollection from '../../../../src/api/current-collection';

describe('Wellcome Genre: Boxes collection', () => {
  const entity = normalize(boxes, collection);
  const state = {
    ...entity.entities,
    config: {
      defaultLanguage: 'en',
    },
    routing: {
      currentCollection: entity.result,
      currentManifest: null,
    },
  };

  it('should have an id', () => {
    expect(currentCollection.getId(state)).toEqual(
      'https://wellcomelibrary.org/service/collections/genres/Boxes/'
    );
  });

  it('should have a type of collection', () => {
    expect(currentCollection.getType(state)).toEqual('sc:Collection');
  });

  it('should have a label', () => {
    expect(currentCollection.getLabel(state)).toEqual([
      {
        '@language': 'en',
        '@value': ' Genre: Boxes',
      },
    ]);
  });

  it('should have members', () => {
    expect(currentCollection.getMemberIds(state)).toEqual([
      'https://wellcomelibrary.org/iiif/b28560577/manifest',
      'https://wellcomelibrary.org/iiif/b28560644/manifest',
      'https://wellcomelibrary.org/iiif/b28668844/manifest',
      'https://wellcomelibrary.org/iiif/b28671624/manifest',
      'https://wellcomelibrary.org/iiif/b28671600/manifest',
    ]);
    const members = currentCollection.getMembers(state);
    expect(members.length).toEqual(5);
    expect(members[0].label).toEqual([
      {
        '@language': 'en',
        '@value':
          'Bio-Light :the 3-day detox diet : lose weight, look good, feel great',
      },
    ]);
  });

  it('should have manifests', () => {
    const manifests = currentCollection.getManifests(state);
    expect(manifests.length).toEqual(5);
    expect(manifests[0].label).toEqual([
      {
        '@language': 'en',
        '@value':
          'Bio-Light :the 3-day detox diet : lose weight, look good, feel great',
      },
    ]);
  });
});
