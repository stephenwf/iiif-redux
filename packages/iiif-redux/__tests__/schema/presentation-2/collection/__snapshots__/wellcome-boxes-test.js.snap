// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Wellcome Genre: Boxes collection should normalize 1`] = `
Object {
  "collections": Object {
    "https://wellcomelibrary.org/service/collections/genres/Boxes/": Object {
      "@context": "http://iiif.io/api/presentation/2/context.json",
      "@id": "https://wellcomelibrary.org/service/collections/genres/Boxes/",
      "@type": "sc:Collection",
      "label": Array [
        Object {
          "@language": "en",
          "@value": " Genre: Boxes",
        },
      ],
      "manifests": Array [
        "https://wellcomelibrary.org/iiif/b28560577/manifest",
        "https://wellcomelibrary.org/iiif/b28560644/manifest",
        "https://wellcomelibrary.org/iiif/b28668844/manifest",
        "https://wellcomelibrary.org/iiif/b28671624/manifest",
        "https://wellcomelibrary.org/iiif/b28671600/manifest",
      ],
    },
  },
  "externalResources": Object {
    "https://wellcomelibrary.org/data/b28560577.json": Object {
      "@id": "https://wellcomelibrary.org/data/b28560577.json",
      "format": "application/json",
      "profile": "http://wellcomelibrary.org/profiles/res",
    },
    "https://wellcomelibrary.org/data/b28560644.json": Object {
      "@id": "https://wellcomelibrary.org/data/b28560644.json",
      "format": "application/json",
      "profile": "http://wellcomelibrary.org/profiles/res",
    },
    "https://wellcomelibrary.org/data/b28668844.json": Object {
      "@id": "https://wellcomelibrary.org/data/b28668844.json",
      "format": "application/json",
      "profile": "http://wellcomelibrary.org/profiles/res",
    },
    "https://wellcomelibrary.org/data/b28671600.json": Object {
      "@id": "https://wellcomelibrary.org/data/b28671600.json",
      "format": "application/json",
      "profile": "http://wellcomelibrary.org/profiles/res",
    },
    "https://wellcomelibrary.org/data/b28671624.json": Object {
      "@id": "https://wellcomelibrary.org/data/b28671624.json",
      "format": "application/json",
      "profile": "http://wellcomelibrary.org/profiles/res",
    },
    "https://wellcomelibrary.org/item/b28560577": Object {
      "@id": "https://wellcomelibrary.org/item/b28560577",
      "format": "text/html",
    },
    "https://wellcomelibrary.org/item/b28560644": Object {
      "@id": "https://wellcomelibrary.org/item/b28560644",
      "format": "text/html",
    },
    "https://wellcomelibrary.org/item/b28668844": Object {
      "@id": "https://wellcomelibrary.org/item/b28668844",
      "format": "text/html",
    },
    "https://wellcomelibrary.org/item/b28671600": Object {
      "@id": "https://wellcomelibrary.org/item/b28671600",
      "format": "text/html",
    },
    "https://wellcomelibrary.org/item/b28671624": Object {
      "@id": "https://wellcomelibrary.org/item/b28671624",
      "format": "text/html",
    },
    "https://wellcomelibrary.org/resource/dublincore/b28560577": Object {
      "@id": "https://wellcomelibrary.org/resource/dublincore/b28560577",
      "format": "application/ld+json",
      "profile": "http://iiif.io/community/profiles/discovery/dc",
    },
    "https://wellcomelibrary.org/resource/dublincore/b28560644": Object {
      "@id": "https://wellcomelibrary.org/resource/dublincore/b28560644",
      "format": "application/ld+json",
      "profile": "http://iiif.io/community/profiles/discovery/dc",
    },
    "https://wellcomelibrary.org/resource/dublincore/b28668844": Object {
      "@id": "https://wellcomelibrary.org/resource/dublincore/b28668844",
      "format": "application/ld+json",
      "profile": "http://iiif.io/community/profiles/discovery/dc",
    },
    "https://wellcomelibrary.org/resource/dublincore/b28671600": Object {
      "@id": "https://wellcomelibrary.org/resource/dublincore/b28671600",
      "format": "application/ld+json",
      "profile": "http://iiif.io/community/profiles/discovery/dc",
    },
    "https://wellcomelibrary.org/resource/dublincore/b28671624": Object {
      "@id": "https://wellcomelibrary.org/resource/dublincore/b28671624",
      "format": "application/ld+json",
      "profile": "http://iiif.io/community/profiles/discovery/dc",
    },
    "https://wellcomelibrary.org/resource/schemaorg/b28560577": Object {
      "@id": "https://wellcomelibrary.org/resource/schemaorg/b28560577",
      "format": "application/ld+json",
      "profile": "http://iiif.io/community/profiles/discovery/schema",
    },
    "https://wellcomelibrary.org/resource/schemaorg/b28560644": Object {
      "@id": "https://wellcomelibrary.org/resource/schemaorg/b28560644",
      "format": "application/ld+json",
      "profile": "http://iiif.io/community/profiles/discovery/schema",
    },
    "https://wellcomelibrary.org/resource/schemaorg/b28668844": Object {
      "@id": "https://wellcomelibrary.org/resource/schemaorg/b28668844",
      "format": "application/ld+json",
      "profile": "http://iiif.io/community/profiles/discovery/schema",
    },
    "https://wellcomelibrary.org/resource/schemaorg/b28671600": Object {
      "@id": "https://wellcomelibrary.org/resource/schemaorg/b28671600",
      "format": "application/ld+json",
      "profile": "http://iiif.io/community/profiles/discovery/schema",
    },
    "https://wellcomelibrary.org/resource/schemaorg/b28671624": Object {
      "@id": "https://wellcomelibrary.org/resource/schemaorg/b28671624",
      "format": "application/ld+json",
      "profile": "http://iiif.io/community/profiles/discovery/schema",
    },
  },
  "imageResources": Object {
    "https://dlcs.io/iiif-img/wellcome/1/19543f4a-a32c-4f4a-a361-47e4247924ce/full/!1024,1024/0/default.jpg": Object {
      "@id": "https://dlcs.io/iiif-img/wellcome/1/19543f4a-a32c-4f4a-a361-47e4247924ce/full/!1024,1024/0/default.jpg",
      "@type": "dctypes:Image",
      "service": Array [
        "https://dlcs.io/iiif-img/wellcome/1/19543f4a-a32c-4f4a-a361-47e4247924ce",
      ],
    },
    "https://dlcs.io/iiif-img/wellcome/1/37b4ea18-0b5f-48ff-b727-cfc759f52f5c/full/!1024,1024/0/default.jpg": Object {
      "@id": "https://dlcs.io/iiif-img/wellcome/1/37b4ea18-0b5f-48ff-b727-cfc759f52f5c/full/!1024,1024/0/default.jpg",
      "@type": "dctypes:Image",
      "service": Array [
        "https://dlcs.io/iiif-img/wellcome/1/37b4ea18-0b5f-48ff-b727-cfc759f52f5c",
      ],
    },
    "https://dlcs.io/iiif-img/wellcome/1/3b2667e0-1f3f-4a51-b85d-b04bc84a0306/full/!1024,1024/0/default.jpg": Object {
      "@id": "https://dlcs.io/iiif-img/wellcome/1/3b2667e0-1f3f-4a51-b85d-b04bc84a0306/full/!1024,1024/0/default.jpg",
      "@type": "dctypes:Image",
      "service": Array [
        "https://dlcs.io/iiif-img/wellcome/1/3b2667e0-1f3f-4a51-b85d-b04bc84a0306",
      ],
    },
    "https://dlcs.io/iiif-img/wellcome/1/708ba0ea-09f7-44d9-a8f3-55f26201bbf9/full/!1024,1024/0/default.jpg": Object {
      "@id": "https://dlcs.io/iiif-img/wellcome/1/708ba0ea-09f7-44d9-a8f3-55f26201bbf9/full/!1024,1024/0/default.jpg",
      "@type": "dctypes:Image",
      "service": Array [
        "https://dlcs.io/iiif-img/wellcome/1/708ba0ea-09f7-44d9-a8f3-55f26201bbf9",
      ],
    },
    "https://dlcs.io/iiif-img/wellcome/1/bbce4247-f975-4324-97aa-f6d41299404e/full/!1024,1024/0/default.jpg": Object {
      "@id": "https://dlcs.io/iiif-img/wellcome/1/bbce4247-f975-4324-97aa-f6d41299404e/full/!1024,1024/0/default.jpg",
      "@type": "dctypes:Image",
      "service": Array [
        "https://dlcs.io/iiif-img/wellcome/1/bbce4247-f975-4324-97aa-f6d41299404e",
      ],
    },
  },
  "manifests": Object {
    "https://wellcomelibrary.org/iiif/b28560577/manifest": Object {
      "@id": "https://wellcomelibrary.org/iiif/b28560577/manifest",
      "@type": "sc:Manifest",
      "label": Array [
        Object {
          "@language": "en",
          "@value": "Bio-Light :the 3-day detox diet : lose weight, look good, feel great",
        },
      ],
      "related": Array [
        "https://wellcomelibrary.org/item/b28560577",
      ],
      "seeAlso": Array [
        "https://wellcomelibrary.org/data/b28560577.json",
        "https://wellcomelibrary.org/resource/schemaorg/b28560577",
        "https://wellcomelibrary.org/resource/dublincore/b28560577",
      ],
      "thumbnail": "https://dlcs.io/iiif-img/wellcome/1/37b4ea18-0b5f-48ff-b727-cfc759f52f5c/full/!1024,1024/0/default.jpg",
    },
    "https://wellcomelibrary.org/iiif/b28560644/manifest": Object {
      "@id": "https://wellcomelibrary.org/iiif/b28560644/manifest",
      "@type": "sc:Manifest",
      "label": Array [
        Object {
          "@language": "en",
          "@value": "Disprin Extra fast and effective : Disprin Asprin fast and effective.",
        },
      ],
      "related": Array [
        "https://wellcomelibrary.org/item/b28560644",
      ],
      "seeAlso": Array [
        "https://wellcomelibrary.org/data/b28560644.json",
        "https://wellcomelibrary.org/resource/schemaorg/b28560644",
        "https://wellcomelibrary.org/resource/dublincore/b28560644",
      ],
      "thumbnail": "https://dlcs.io/iiif-img/wellcome/1/bbce4247-f975-4324-97aa-f6d41299404e/full/!1024,1024/0/default.jpg",
    },
    "https://wellcomelibrary.org/iiif/b28668844/manifest": Object {
      "@id": "https://wellcomelibrary.org/iiif/b28668844/manifest",
      "@type": "sc:Manifest",
      "label": Array [
        Object {
          "@language": "en",
          "@value": "Coins from friends of the Foodchain :The Food Chain, feeding the fight against HIV and AIDS",
        },
      ],
      "related": Array [
        "https://wellcomelibrary.org/item/b28668844",
      ],
      "seeAlso": Array [
        "https://wellcomelibrary.org/data/b28668844.json",
        "https://wellcomelibrary.org/resource/schemaorg/b28668844",
        "https://wellcomelibrary.org/resource/dublincore/b28668844",
      ],
      "thumbnail": "https://dlcs.io/iiif-img/wellcome/1/708ba0ea-09f7-44d9-a8f3-55f26201bbf9/full/!1024,1024/0/default.jpg",
    },
    "https://wellcomelibrary.org/iiif/b28671600/manifest": Object {
      "@id": "https://wellcomelibrary.org/iiif/b28671600/manifest",
      "@type": "sc:Manifest",
      "label": Array [
        Object {
          "@language": "en",
          "@value": "Questions about AIDS :call us at National AIDS Helplines : take a free card : phone free - in confidence",
        },
      ],
      "related": Array [
        "https://wellcomelibrary.org/item/b28671600",
      ],
      "seeAlso": Array [
        "https://wellcomelibrary.org/data/b28671600.json",
        "https://wellcomelibrary.org/resource/schemaorg/b28671600",
        "https://wellcomelibrary.org/resource/dublincore/b28671600",
      ],
      "thumbnail": "https://dlcs.io/iiif-img/wellcome/1/3b2667e0-1f3f-4a51-b85d-b04bc84a0306/full/!1024,1024/0/default.jpg",
    },
    "https://wellcomelibrary.org/iiif/b28671624/manifest": Object {
      "@id": "https://wellcomelibrary.org/iiif/b28671624/manifest",
      "@type": "sc:Manifest",
      "label": Array [
        Object {
          "@language": "en",
          "@value": "Questions about HIV and AIDS :call us at National AIDS Helplines",
        },
      ],
      "related": Array [
        "https://wellcomelibrary.org/item/b28671624",
      ],
      "seeAlso": Array [
        "https://wellcomelibrary.org/data/b28671624.json",
        "https://wellcomelibrary.org/resource/schemaorg/b28671624",
        "https://wellcomelibrary.org/resource/dublincore/b28671624",
      ],
      "thumbnail": "https://dlcs.io/iiif-img/wellcome/1/19543f4a-a32c-4f4a-a361-47e4247924ce/full/!1024,1024/0/default.jpg",
    },
  },
  "services": Object {
    "https://dlcs.io/iiif-img/wellcome/1/19543f4a-a32c-4f4a-a361-47e4247924ce": Object {
      "@context": "http://iiif.io/api/image/2/context.json",
      "@id": "https://dlcs.io/iiif-img/wellcome/1/19543f4a-a32c-4f4a-a361-47e4247924ce",
      "profile": "http://iiif.io/api/image/2/level1.json",
    },
    "https://dlcs.io/iiif-img/wellcome/1/37b4ea18-0b5f-48ff-b727-cfc759f52f5c": Object {
      "@context": "http://iiif.io/api/image/2/context.json",
      "@id": "https://dlcs.io/iiif-img/wellcome/1/37b4ea18-0b5f-48ff-b727-cfc759f52f5c",
      "profile": "http://iiif.io/api/image/2/level1.json",
    },
    "https://dlcs.io/iiif-img/wellcome/1/3b2667e0-1f3f-4a51-b85d-b04bc84a0306": Object {
      "@context": "http://iiif.io/api/image/2/context.json",
      "@id": "https://dlcs.io/iiif-img/wellcome/1/3b2667e0-1f3f-4a51-b85d-b04bc84a0306",
      "profile": "http://iiif.io/api/image/2/level1.json",
    },
    "https://dlcs.io/iiif-img/wellcome/1/708ba0ea-09f7-44d9-a8f3-55f26201bbf9": Object {
      "@context": "http://iiif.io/api/image/2/context.json",
      "@id": "https://dlcs.io/iiif-img/wellcome/1/708ba0ea-09f7-44d9-a8f3-55f26201bbf9",
      "profile": "http://iiif.io/api/image/2/level1.json",
    },
    "https://dlcs.io/iiif-img/wellcome/1/bbce4247-f975-4324-97aa-f6d41299404e": Object {
      "@context": "http://iiif.io/api/image/2/context.json",
      "@id": "https://dlcs.io/iiif-img/wellcome/1/bbce4247-f975-4324-97aa-f6d41299404e",
      "profile": "http://iiif.io/api/image/2/level1.json",
    },
  },
}
`;
